src/picasso-definition/interactions/pan.js

```diff
@@ -48,11 +49,24 @@ const pan = ({ chart, actions, viewHandler, rtl }) => ({
         return false;
       }
 
-      [this.pointAreaPanned] = chart
+      this.area = chart
         .componentsFromPoint({ x: e.center.x, y: e.center.y })
         .filter((c) => c.key === KEYS.COMPONENT.POINT || c.key === KEYS.COMPONENT.HEAT_MAP);
 
-      return this.pointAreaPanned;
+      if (!this.area.length) {
+        return false;
+      }
+
+      const rectSize = this.area[0]?.rect?.computedPhysical;
+
+      if (rectSize?.height && rectSize?.width) {
+        this.componentSize = rectSize;
```

Comment:
it will be safer if we do this.componentSize = {width: rectSize.width, height: rectSize.height};

Reply:
Or better to do the following: 

if (rectSize?.height && rectSize?.width) {
       lastRectSize = rectSize;
 }
this.componentSize = lastRectSize;